runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java

```diff
@@ -61,6 +61,7 @@
             .expireAfterAccess(runtime.getParameters().getCacheExpiryTime(), TimeUnit.SECONDS)
             .expireAfterWrite(runtime.getParameters().getCacheExpiryTime(), TimeUnit.SECONDS)
             .recordStats()
+            .concurrencyLevel(Runtime.getRuntime().availableProcessors() * 2)
```

Comment:
Can you please explain how this is related to NoRollBackException fix?

Reply:
This addresses the side effect of locking. Internally, the cache partitions the address space into buckets and each bucket has a reader/writer lock. The size of these buckets (i.e. key-partition space size) affects concurrency on the global cache. For example, if there's one bucked then there's only serial access to the cache. Conversely, if there's as many locks as keys then the cache is fully parallel (with respect to key access). In essence, this is a trade-off between memory usage and concurrency and needs to be tuned, setting CPU * 2 is just a "reasonable guess".